<launch>
    
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/localization.rviz" />

    <!--- TF -->
    <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10" />
    <node pkg="tf" type="static_transform_publisher" name="odom_to_base" args="0 0 0 -1.570795 -1.570795 0         /laser_odom /base_link   10" />
    <node pkg="tf" type="static_transform_publisher" name="base_to_velo" args="0 0 0 0 0 0        /base_link /velodyne   10" />

    <node pkg="nodelet" type="nodelet" args="manager"
    name="legoloam_nodelet_manager" output="screen" />
	<node pkg="nodelet" type="nodelet" args="load lego_loam/ImageProjection legoloam_nodelet_manager"
    name="ImageProjection_loader" output="screen">
    <!-- <remap from="velodyne_points" to="points_raw"/> -->
    </node>	
	<node pkg="nodelet" type="nodelet" args="load lego_loam/FeatureAssociation legoloam_nodelet_manager"
    name="FeatureAssociation_loader" output="screen"/>

    <!--- Initialize -->    
    <node pkg="nodelet" type="nodelet" args="load lego_loam/GlobalLocalization legoloam_nodelet_manager"
    name="GlobalLocalization_loader" output="screen">
        <param name="model_path" value="c:\opt\ros\melodic\test_ws\src\global-LeGO-LOAM\train\locnet_descriptor510.pt"/>
        <param name="feature_cloud_path" value="C:\Users\Haeyeon Kim\Desktop\lego_loam_result\feature_cloud.pcd"/>
        <param name="search_radius" value="1000.0"/>
    </node>

    <node pkg="nodelet" type="nodelet" args="load lego_loam/ParticleFilter3D legoloam_nodelet_manager"
    name="ParticleFilter3D_loader" output="screen">
        <param name="num_particles" value="5000"/>
        <param name="sampling_covariance" value="0.5"/> <!--Initial convariance-->
        
        <param name="skip_measure" value="1"/>
        <param name="clip_near" value="2.0"/>
        <param name="clip_far" value="10.0"/>
        <param name="clip_z_min" value="-1.0"/>
        <param name="clip_z_max" value="10.0"/>

        <param name="resample_var_x" value="0.1"/>
        <param name="resample_var_y" value="0.1"/>
        <param name="resample_var_yaw" value="0.1"/>
        <param name="max_search_radius" value="0.5"/>
        <param name="min_z_pose" value="0.0"/>
        <param name="max_z_pose" value="2.0"/>

        <param name="expansion_var_x" value="1.0"/>
        <param name="expansion_var_y" value="1.0"/>
        <param name="expansion_var_yaw" value="0.5"/>

        <param name="map_downsample_x" value="0.5"/>
        <param name="map_downsample_y" value="0.5"/>
        <param name="map_downsample_z" value="0.5"/>
        <param name="map_chunk" value="1000"/>
        
        <param name="downsample_x" value="0.1"/>
        <param name="downsample_y" value="0.1"/>
        <param name="downsample_z" value="0.1"/>
        <param name="match_ratio_thresh" value="0.8"/>     
    </node>

</launch>

